http-clients/src/main/java/com/palantir/remoting/http/UserAgents.java

```diff
@@ -30,13 +28,20 @@ public static String getUserAgent(String userAgent, String version) {
 
     /**
      * Constructs a user agent from the {@link java.util.jar.Attributes.Name#IMPLEMENTATION_TITLE} and {@link
-     * java.util.jar.Attributes.Name#IMPLEMENTATION_VERSION} of the package of the provided class. The default value for
-     * both properties is "{@value DEFAULT_VALUE}".
+     * java.util.jar.Attributes.Name#IMPLEMENTATION_VERSION} of the package of the provided class.
+     * <p>
+     * Typically, the title and version are extracted from {@code MANIFEST.MF} entries of the Jar package containing the
+     * given class.
+     *
+     * @throws IllegalArgumentException if the implementation title or version are not provided in the class's {@link
+     * Package}
      */
     public static String fromClass(Class<?> clazz) {
         Package classPackage = clazz.getPackage();
-        String userAgent = Optional.fromNullable(classPackage.getImplementationTitle()).or(DEFAULT_VALUE);
-        String version = Optional.fromNullable(classPackage.getImplementationVersion()).or(DEFAULT_VALUE);
+        String userAgent = classPackage.getImplementationTitle();
+        String version = classPackage.getImplementationVersion();
+        checkArgument(userAgent != null, "Cannot load user agent from implementation title");
```

Comment:
sorry to be a pain, but I think these should say `...from implementation title for class: %s", clazz)`